src/lib.rs

```diff
@@ -84,25 +83,27 @@ fn join_with_or<TItem: fmt::Display>(list: &[TItem]) -> String {
     ret
 }
 
-struct CachedQueries(HashMap<SupportedLanguageName, OnceLock<Result<Arc<Query>, QueryError>>>);
+#[derive(Default)]
+struct CachedQueries(BySupportedLanguage<OnceLock<Result<Arc<Query>, QueryError>>>);
 
 impl CachedQueries {
     fn get_and_cache_query_for_language(
         &self,
         query_source: &str,
         language: SupportedLanguage,
     ) -> Option<Arc<Query>> {
-        self.0
-            .get(&language.name)
-            .unwrap()
-            .get_or_init(|| maybe_get_query(query_source, language.language).map(Arc::new))
+        self.0[language]
```

Comment:
So I guess this was one of the goals, getting rid of the "fallible lookup" (ie the `.unwrap()`)